File: sms_back_end/database/multi_tenant_school_management/schemas/teacher.py
from typing import Optional
from uuid import UUID

from pydantic import BaseModel

from database.multi_tenant_school_management.schemas.user import UserCreate, UserRead


class TeacherBase(BaseModel):
    school_id: UUID


class TeacherCreate(TeacherBase):
    user: UserCreate


class TeacherRead(TeacherBase):
    id: UUID
    user: UserRead

    model_config = {
        "from_attributes": True
    }

File: sms-frontend/types/teacher.ts
import {UserCreate, UserRead} from "@/types/user";
import {UUID} from "@/types/shared.primitive";

export interface TeacherBase {
    school_id: UUID;
}

export interface TeacherCreate extends TeacherBase {
    user: UserCreate;
}

export interface TeacherRead extends TeacherBase {
    id: UUID;
    user: UserRead;
}